NineAnimator/Models/Anime Source/animesaturn.com/AnimeSaturn+Anime.swift

```diff
@@ -0,0 +1,92 @@
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with NineAnimator.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+import Foundation
+import SwiftSoup
+
+extension NASourceAnimeSaturn {
+    func anime(from link: AnimeLink) -> NineAnimatorPromise<Anime> {
+        self.requestManager
+            .request(url: link.link, handling: .browsing)
+            .responseString
+            .then {
+                responseContent -> Anime in
+                let bowl = try SwiftSoup.parse(responseContent)
+                let animeTitle = try bowl.select("div.container.anime-title-as.mb-3.w-100 b").text()
+                let animeArtworkUrl = URL(
+                    string: try bowl.select(".cover>img").attr("src")
+                    ) ?? link.image
+                let reconstructedAnimeLink = AnimeLink(
+                    title: animeTitle,
+                    link: link.link,
+                    image: animeArtworkUrl,
+                    source: self
+                )
+                
+                // Obtain the list of episodes
+                let episodes = try bowl.select("div.tab-content div div.episodes-button").reduce(into: [EpisodeLink]()) {
+                    collection, container in
+                    let name =  try container.select("a").text().components(separatedBy: " ")
+                    let episodeName = name[1]
+                    var episodeLink = try container.select("a").attr("href")
+                    episodeLink = episodeLink.replacingOccurrences(of: "'", with: "\'")
+                    if !episodeLink.isEmpty {
+                        collection.append(.init(
+                            identifier: episodeLink,
+                            name: episodeName,
+                            server: NASourceAnimeSaturn.AnimeSaturnStream,
+                            parent: reconstructedAnimeLink
+                            ))
+                    }
+                }
+                
+                // Information
+                let alias = try bowl.select("div.box-trasparente-alternativo.rounded").first()?.text()
+                let animeSynopsis = try bowl.select("#shown-trama").text()
+                //var additionalAttributes = [Anime.AttributeKey: Any]()
+                // Attributes
+                let additionalAttributes = try bowl.select("div.container.shadow.rounded.bg-dark-as-box.mb-3.p-3.w-100.text-white").reduce(into: [Anime.AttributeKey: Any]()) { attributes, entry in
+                    let info = entry.debugDescription.components(separatedBy: "<br>")
+                    for elem in info {
+                        if elem.contains("<b>Voto:</b> ") {
+                            var rat = elem.dropFirst(13).components(separatedBy: "<b>Voto:</b> ")
+                            rat = rat[1].components(separatedBy: "/")
```

Comment:
Array index access is not recommended. Any changes on the source website can potentially crash the app.
* L68: `elem.dropFirst(13)`
* L69: `rat[1]`
* L70: `rat[0]`
* L76: `rat[1]`

Reply:
If your forced to use index access, you could use the new `[safe: 1]` collection extension that I have made: https://github.com/SuperMarcus/NineAnimator/blob/master/NineAnimator/Utilities/Array%2BSafeSubscript.swift
It returns nil if you try to access index that is out of bounds instead of crashing app.

Reply:
yeah, I have to use it. because I have no idea how to get the rating rappresented 4.5/5 without separate the string by "/"